### DISCUSSION TITLE
Welcome to Node.js-Design-Patterns-Third-Edition Discussions!

### DISCUSSION THREAD

**Discussion by lmammino** [PacktPublishing/Node.js-Design-Patterns-Third-Edition#23] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Jack-Barry** [comment]
I'm a self-taught programmer trying to get a more solid understanding of the fundamentals of NodeJS and computer programming/computer science in general. I'll be trying to spark some conversation around the topics in this book when I get some free time.

I've got a lot to learn so if you see or hear me doing something stupid, feel free to provide constructive criticism

**Reply by lmammino** [reply]
Hey @Jack-Barry, really nice to meet you!

We look forward to seeing you opening some discussion. Also, our philosophy is that there is no _stupid question_. So feel free to ask any question that makes sense to you. We will be more than happy to help. We have been learning by asking tons of questions to people with more experience than us, so now it's hopefully our chance to give something back!

**Comment by arvi** [comment]
Hello. I learned about this book which was recommended by another book I just finished called "JavaScript in the Industry".

I would like to know if the lessons in this book are not yet outdated now that Node version is 16. Would you still recommend this book in 2022? Also, in case there are major updates, will previous buyers get updated version or have to buy again?

Thank you.

**Reply by lmammino** [reply]
Hello @arvi and thanks for your interest in Node.js Design Patterns.

Of course as one of the authors I am a bit biased, but I would say that there aren't major changes in Node.js 16 that would make this book out of date.

The book is not focused on specific Node.js feature but more on illustrating major design patterns and techniques that have generally a longer timespan than Node.js releases.

With that being said, all the examples in the book are already using very modern features (ESM Modules, Async/Await, modern features like async iterators, etc.). When we wrote the third edition with put some special attention in trying to make sure that the book would be as future proof as possilble :)

Let us know what you think and of course don't hesitate to ask more questions!

All the best

- Luciano

**Comment by OpenTechConsult** [comment]
Hello Everyone, 

I'm Sandro. I am a web developer and I would like to a be Node.js backend developer. I am here to interact with the authors and also with the people in this community. I would also love to help the community if possible. I am a junior software ingeneer.
